contracts/VestingEscrowFullyRevokable.vy

```diff
@@ -0,0 +1,375 @@
+# @version 0.3.7
+
+"""
+@title Vesting Escrow
+@author Curve Finance, Yearn Finance, Lido Finance
+@license MIT
+@notice Vests ERC20 tokens for a single address
+@dev Intended to be deployed many times via `VotingEscrowFactory`
+"""
+
+from vyper.interfaces import ERC20
+
+
+event Activated:
+    recipient: indexed(address)
+    amount: uint256
+    owner: address
+    manager: address
+
+
+event Claim:
+    recipient: indexed(address)
+    beneficiary: address
+    claimed: uint256
+
+
+event RevokeUnvested:
+    recipient: indexed(address)
+    revoked: uint256
+
+
+event RevokeAll:
+    recipient: indexed(address)
+    revoked: uint256
+
+
+event OwnerChanged:
+    owner: address
+
+
+event ManagerChanged:
+    manager: address
+
+
+event ERC20Recovered:
+    token: address
+    amount: uint256
+
+
+event VotingAdapterUpdated:
+    addr: address
+
+
+LIDO_VOTING_CONTRACT_ADDR: immutable(address)
+SNAPSHOT_DELEGATE_CONTRACT_ADDR: immutable(address)
+
+recipient: public(address)
+token: public(ERC20)
+start_time: public(uint256)
+end_time: public(uint256)
+cliff_length: public(uint256)
+total_locked: public(uint256)
+total_claimed: public(uint256)
+disabled_at: public(uint256)
+is_fully_revoked: public(bool)
+initialized: public(bool)
+activated: public(bool)
+voting_adapter_addr: public(address)
+
+owner: public(address)
+manager: public(address)
+
+
+@external
+def __init__(voting_addr: address, snapshot_addr: address):
+    """
+    @notice Initialize source contract implementation.
+    @param voting_addr Address of the Lido Voting contract
+    @param snapshot_addr Address of the Shapshot Delegate contract
+    """
+    # ensure that the original contract cannot be initialized
+    self.initialized = True
```

Comment:
INteresting but does it make sense?

Reply:
Yep. This is crucial to disallow usage of the implementation itself as escrow